src/rust/sqs-executor/src/redis_cache.rs

```diff
@@ -107,26 +118,109 @@ impl RedisCache {
         }
     }
 
+    async fn _get_all<CA>(
+        &mut self,
+        cacheables: Vec<CA>,
+    ) -> Result<Vec<CacheResponse>, RedisCacheError>
+    where
+        CA: Cacheable + Send + Sync + 'static,
+    {
+        let cacheable_responses: Vec<(CA, Option<CacheResponse>)> = {
+            let mut lru_cache = self.lru_cache.lock().unwrap();
+
+            cacheables
+                .into_iter()
+                .map(|cacheable| {
+                    // if this hits, it's a Some(CacheResponse::Hit)
+                    // otherwise, we use None to signify that we should check in with redis
+                    let identity_bytes = cacheable.identity();
+                    let lru_cache_response =
+                        lru_cache.get(&identity_bytes).map(|_| CacheResponse::Hit);
+
+                    (cacheable, lru_cache_response)
+                })
+                .collect()
+        };
+
+        // if the LRU cache satisfied all our needs, we should return early
+        let are_all_keys_handled = cacheable_responses
+            .iter()
+            .fold(true, |acc, (_, response)| acc && response.is_some());
+
+        if are_all_keys_handled {
+            return Ok(cacheable_responses
+                .into_iter()
+                .filter_map(|(_, response)| response)
+                .collect());
+        }
+
+        // fetch the responses from redis and convert them into CacheResponses
+        // this Vec<_> should be equal in size to the number of entries in cacheable_responses with
+        // 'response' set to None
+        let mut unknown_cacheable_responses: VecDeque<CacheResponse> = {
+            // this vec will be len() > 0 since we would've returned earlier otherwise
+            let unknown_cacheables: Vec<_> = cacheable_responses
+                .iter()
+                .filter(|(_, response)| response.is_none())
+                .map(|(cacheable, _)| cacheable.identity())
+                .collect();
+
+            let client_pool = self.connection_pool.clone();
+            let mut client = client_pool.get().await;
+
+            client
+                .mget(&unknown_cacheables)
+                .timeout(Duration::from_millis(300))
```

Comment:
I'm okay with increasing the `Duration` value here, but in general yea it could time out and we'd return saying "everything missed". Colin was fine with this when I brought it up, but if you ahve a suggestion as an alternative I'm all ears. I was going to fail early and return a vec with misses here that was the appropriate size back here.